go-controller/hybrid-overlay/pkg/controller/node.go

```diff
@@ -0,0 +1,80 @@
+package controller
+
+import (
+	"bytes"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/util"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn"
+
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/klog"
+)
+
+// nodeChanged returns true if any relevant node attributes changed
+func nodeChanged(node1 *kapi.Node, node2 *kapi.Node) bool {
+	cidr1, nodeIP1, drMAC1 := getNodeDetails(node1)
+	cidr2, nodeIP2, drMAC2 := getNodeDetails(node2)
+
+	if cidr1 != nil && cidr2 != nil && nodeIP1 != nil && nodeIP2 != nil && drMAC1 != nil && drMAC2 != nil {
```

Comment:
In this case, you don't really care if it is nil or not you just care if there is a difference between the 2. So you can simplify all this by using returning the output of !reflect.DeepEqual

Reply:
Simpler perhaps, but quite a bit more expensive...

Reply:
I don't think the cost is that concerning, but it's only a few lines. If you would rather leave it then I'm fine with it.

Reply:
changed